src/cargo/core/compiler/timings/mod.rs

```diff
@@ -139,6 +139,29 @@ impl<'gctx> Timings<'gctx> {
         let report_json = has_report(TimingOutput::Json);
         let enabled = report_html | report_json | bcx.logger.is_some();
 
+        let mut timings = Timings {
+            gctx: bcx.gctx,
+            enabled,
+            report_html,
+            report_json,
+            start: bcx.gctx.creation_time(),
+            start_str: String::new(),
+            root_targets: Vec::new(),
+            profile: String::new(),
+            total_fresh: 0,
+            total_dirty: 0,
+            unit_times: Vec::new(),
+            active: HashMap::new(),
+            concurrency: Vec::new(),
+            last_cpu_state: None,
+            last_cpu_recording: Instant::now(),
+            cpu_usage: Vec::new(),
+        };
```

Comment:
downside is we can't track if we forgot to initialize a new field

For most of the context, should we move that to an inner type and have an `Option` around it?

Or move the creation of this into `if !enabled`?

Reply:
Fixed in <https://github.com/rust-lang/cargo/compare/4aae037f291bebe52ab9b8edd4833e9302ea72dc..1ce4adb2359fba3344c6ee6bdc73cbaf5cf315d9>